modules/appeals_api/spec/models/notice_of_disagreement_spec.rb

```diff
@@ -286,6 +287,76 @@
       end
     end
 
+    context 'when required claimant data is missing' do
+      context 'when data provided but headers missing' do
+        let(:auth_headers) { fixture_as_json 'valid_10182_headers.json', version: 'v2' }
+        let(:form_data) { fixture_as_json 'valid_10182_extra.json', version: 'v2' }
+        let(:notice_of_disagreement_v2) do
+          described_class.new(api_version: 'v2', form_data: form_data, auth_headers: auth_headers,
+                              board_review_option: 'hearing')
+        end
+
+        it 'creates an invalid record' do
+          expect(notice_of_disagreement_v2.valid?).to be false
+          expect(notice_of_disagreement_v2.errors.to_a.length).to eq 1
+          expect(notice_of_disagreement_v2.errors.to_a.first).to include 'missing claimant headers'
+        end
+      end
+
+      context 'when headers provided but missing data' do
+        let(:auth_headers) { fixture_as_json 'valid_10182_headers_extra.json', version: 'v2' }
+        let(:form_data) { fixture_as_json 'valid_10182.json', version: 'v2' }
+        let(:notice_of_disagreement_v2) do
+          described_class.new(api_version: 'v2', form_data: form_data, auth_headers: auth_headers,
+                              board_review_option: 'hearing')
+        end
+
+        it 'creates an invalid record' do
+          expect(notice_of_disagreement_v2.valid?).to be false
+          expect(notice_of_disagreement_v2.errors.to_a.length).to eq 1
+          expect(notice_of_disagreement_v2.errors.to_a.first).to include 'missing claimant data'
+        end
+      end
+    end
+
+    context 'bad contestable issue dates' do
+      let(:default_form_data) { fixture_as_json 'valid_10182.json', version: 'v2' }
+      let(:auth_headers) { fixture_as_json 'valid_10182_headers.json', version: 'v2' }
+
+      let(:form_data) do
+        {
+          'data' => default_form_data['data'],
+          'included' => [
+            {
+              'type' => 'contestableIssue',
+              'attributes' => {
+                'issue' => 'PTSD',
+                'decisionDate' => (Time.zone.today + 2).to_s
+              }
+            },
+            {
+              'type' => 'contestableIssue',
+              'attributes' => {
+                'issue' => 'right knee',
+                'decisionDate' => '1901-01-31'
+              }
+            }
+          ]
+        }
+      end
+
+      let(:notice_of_disagreement_v2) do
+        described_class.new(api_version: 'v2', form_data: form_data, auth_headers: auth_headers,
```

Comment:
Example of manually creating and remembering to add the api_version. When forgotten, it might lead the tester to think the guard clauses need to check for api_version.

Reply:
This could be built via one of our factories so that the api version is included by default. Is there a reason we aren't using one here and need to use the `described_class`?

Reply:
Yes, because, because of the way the form data and auth headers are set up. I could make very specific factories for the different test scenarios, but it seems overkill.

I could build the existing factories and modify them, which seems like a fine solution, and I'm happy to refactor, but I still think it's a good example of potential confusing, as it may not be us in the future working on the code...